src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,157 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
+    private static class Node<T> {
+        private T item;
+        private Node<T> next;
+        private Node<T> prev;
+
+        Node(Node<T> prev, T value, Node<T> next) {
+            this.item = value;
+            this.next = next;
+            this.prev = prev;
+        }
+    }
+
+    public MyLinkedList() {
+        first = null;
+        last = null;
+    }
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            addFirst(value);
+        } else {
+            addToEnd(value);
+        }
     }
 
     @Override
     public void add(T value, int index) {
+        if (size == 0 & index == 0) {
+            addFirst(value);
+        } else if (index == 0) {
+            addToStart(value);
+        } else if (index > 0 & index < size) {
+            insert(value, index);
+        } else if (index == size) {
+            addToEnd(value);
+        } else {
+            throw new IndexOutOfBoundsException("Index out of list size!!! Size = " + size);
+        }
     }
```

Comment:
1. Why don't you use double logical operators?
2. You already check your index and throw an exception in method "checkIndex(int index)"